enum.go

```diff
@@ -12,6 +12,11 @@ type Member[T comparable] struct {
 	Value T
 }
 
+// Matcher provides the customr matcher for value type.
+type Matcher[V comparable] interface {
+	Match(v V) bool
```

Comment:
Nitpick: maybe, calling it `Equal` would be better? My first association with `Match` is pattern matching.

Reply:
Agree. I think `Equal` is a more appropriate function name for Value Object. In that case, should the interface name be `Equaler`? Are there any other good names?

Reply:
`Equaler` is fine 👍🏿

Reply:
Just `Equal` is also good. I don't mind breaking conventions if it sounds better. In Rust, traits are typically called after the method they implement. For example, it has `Read` instead of `Reader`. But probably sticking to Go conventions is better.